WordPress/Classes/Services/PostRepository.swift

```diff
@@ -81,6 +81,9 @@ final class PostRepository {
             return
         }
 
+        let status = try await coreDataStack.performQuery { try $0.existingObject(with: postID).status }
+        assert(status == .trash, "This function can only be used to delete trashed posts/pages.")
+
```

Comment:
At first I was surprised to see the `assert` here and not at the start of the method. But I guess the reason is that the code above recur to find the original post, and for the same reason we want to delete the original, we want to check the status on it. Am I reading this right?

Reply:
Yes. This status check doesn't exist in the original delete function in `PostService`. I added it here to catch incorrect calls to this function (which may never happen).

Reply:
👍